Experience Fragments Smoke Tests

stefangrimm opened the pull request:
- Added tests for creating and deleting variations
- Cleaned up Content Package
- Created one single test class for Create/Delete XF/variation to improve performance

Note: I've kept the different steps to be able to revert them if necessary. Easiest way to review is to check the end result.

stefangrimm committed:
CQ-4279630 - [FluidXP] Add Sites FluidXP sanity tests in ESP

Add tests for variations.

stefangrimm committed:
CQ-4279630 - [FluidXP] Add Sites FluidXP sanity tests in ESP

Refactor existing tests.

stefangrimm committed:
CQ-4279630 - [FluidXP] Add Sites FluidXP sanity tests in ESP

Widen filter for better clean up.

stefangrimm committed:
CQ-4279630 - [FluidXP] Add Sites FluidXP sanity tests in ESP

Execute variation-related tests with admin user. Also removed the
non-admin user from the content package.

stefangrimm committed:
CQ-4279630 - [FluidXP] Add Sites FluidXP sanity tests in ESP

Reduce leftovers

stefangrimm committed:
CQ-4279630 - [FluidXP] Add Sites FluidXP sanity tests in ESP

Code cleanup
CQ-4279630 - [FluidXP] Add Sites FluidXP sanity tests in ESP

Create one class to contain the basic CRD tests (performance
improvement)

volteanu commented on xf-smoke/src/main/java/com/adobe/cq/cloud/testing/it/xf/smoke/Constants.java:
```diff
@@ -0,0 +1,35 @@
+
+package com.adobe.cq.cloud.testing.it.xf.smoke;
+
+final class Constants {
```
I think this class doesn't bring much value, each test class can define its own values and it's just a coincidence the values are the same.

stefangrimm replied:
I've removed the `Constants` class. It turned out that the content package is not even required for `XfSmokeIT`, so I'm fine with the suggested change (no duplication in this case).

volteanu reviewed (approved):
LGTM overall!

stefangrimm committed:
CQ-4279630 - [FluidXP] Add Sites FluidXP sanity tests in ESP

Remove Constants class as per feedback.

stefangrimm committed:
CQ-4279630 - [FluidXP] Add Sites FluidXP sanity tests in ESP

Remove superfluous content; avoid installing the CP for XfSmokeIT.

stefangrimm committed:
CQ-4279630 - [FluidXP] Add Sites FluidXP sanity tests in ESP

Obscure content roots to minimize surface for clashes with customer
content.

stefangrimm commented:
I have also reviewed the content package once more and changed the following:

- Removed the `we-retail` labelled content, as it is not required by these tests.
- Added a suffix to the filter roots to make the surface for clashes with customer content as small as possible.

Note that if we would get rid of the `ConvertToXfIT`, we could get rid of the content package as well. IMHO `ConvertToXfIT` is not a typical smoke test.

WDYT?

Note that I didn't set up a Jenkins task, but have run the tests multiple times against a Skyline instance and saw no flakiness. Also, the original (= unadapted) tests are regularly run as part of our internal "Sanity test" Jenkins jobs. If that's not good enough, I guess @tyge68 could set up such a task.

As I am on PTO for the next ~4 weeks, I'll have to hand this over to @martinischeery now, for dispatching it further. Might be necessary to create another fork, unfortunately.

martinischeery commented:
@volteanu Could you please merge this one?

dulvac commented on xf-smoke/src/main/resources/test-content/jcr_root/content/.content.xml:
```diff
@@ -26,5 +26,5 @@
     <geometrixx/>
```
@stefangrimm  Can we remove all these folders? geometrixx, phonegap, social, we-retail etc? They don't make any sense on skyline and I'm not sure why they were there in the first place.

dulvac reviewed (commented):
@stefangrimm , I see some references to some unneeded (?) folders like geometrixx

dulvac commented:
@martinischeery , @tyge68 , if we can get rid of the content package by removing the convert xf it, I would think that makes sense! I agree with @stefangrimm it's not really a smoke test.

@martinischeery , can someone from your team take care of that interruptively?

adobre90 committed:
CQ-4279630 - [FluidXP] Add Sites FluidXP sanity tests in ESP

Removing references of unused folders inside
the jcr_root resource nodes.

Signed-off-by: Antonel Dobre <adobre@adobe.com>

adobre90 committed:
CQ-4279630 - [FluidXP] Add Sites FluidXP sanity tests in ESP

Removing ConverttoXfIT test due to it
not being eligible as a smoke test.

Removing content package as it is not needed anymore.

Signed-off-by: Antonel Dobre <adobre@adobe.com>

dobretony commented:
As per recommendation I've removed ConvertToXfIT and as such managed to get rid of all the content.
I've verified on the FluidXP skyline instances and 6 tests are run without issue ( Create XF, Create XF in Folder, Delete XF, Create Variation, Delete Variation and Delete Master Variation ).

@dulvac @volteanu Could you take a look again?

volteanu reviewed (approved):
Nice!

volteanu commented:
thanks @stefangrimm and @dobretony ! Your changes were released in: https://github.com/adobe/aem-test-samples/tree/com.adobe.cq.cloud.testing.it.xf.smoke-0.2.0
